src/lib/pubkey/curve448/curve448_scalar.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * Ed448 Scalar
+ * (C) 2024 Jack Lloyd
+ *     2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+#include <botan/internal/curve448_scalar.h>
+
+#include <botan/internal/mp_core.h>
+
+namespace Botan {
+
+namespace {
+constexpr size_t WORDS_REDUCE_SZ = words_for_bits(114 * 8);
+constexpr size_t WORDS_C = words_for_bits(28 * 8);
+
+/// @return (q,r) so that x = q*2^446 + r, r < L
+template <size_t S>
+auto div_mod_2_446(std::span<const word, S> x) {
+   if constexpr(S < WORDS_446) {
+      std::array<word, WORDS_446> r = {0};
+      copy_mem(std::span(r).template first<S>(), x);
+      return std::make_pair(std::array<word, 1>({0}), r);
+   } else {
+      std::array<word, WORDS_446> r;
+      copy_mem(r, std::span(x).template first<WORDS_446>());
+      // Clear the two most significant bits
+      r[WORDS_446 - 1] &= ~(word(0b11) << (sizeof(word) * 8 - 2));
+
+      constexpr size_t word_shift = 446 / (sizeof(word) * 8);
+      constexpr size_t bit_shift = 446 % (sizeof(word) * 8);
+
+      std::array<word, S - WORDS_446 + 1> q;
+      bigint_shr2(q.data(), x.data(), x.size(), word_shift, bit_shift);
+
+      return std::make_pair(q, r);
+   }
+}
+
+/// @return a word array for c = 0x8335dc163bb124b65129c96fde933d8d723a70aadc873d6d54a7bb0d
+consteval std::array<word, WORDS_C> c_words() {
+   // Currently load_le does not work with constexpr. Therefore, we have to use this workaround.
+   const std::array<uint8_t, WORDS_C * sizeof(word)> c_bytes{0x0d, 0xbb, 0xa7, 0x54, 0x6d, 0x3d, 0x87, 0xdc, 0xaa, 0x70,
+                                                             0x3a, 0x72, 0x8d, 0x3d, 0x93, 0xde, 0x6f, 0xc9, 0x29, 0x51,
+                                                             0xb6, 0x24, 0xb1, 0x3b, 0x16, 0xdc, 0x35, 0x83};
+   return load_le<std::array<word, WORDS_C>>(c_bytes);
+}
```

Comment:
In a blunt nerd-sniping attempt: if we would manage to make `Botan::hex_decode()` constexpr, this could be:

```suggestion
consteval std::array<word, WORDS_C> c_words() {
   return load_le<std::array<word, WORDS_C>>(Botan::hex_decode(
      "0dbba7546d3d87dcaa703a728d3d93de6fc92951b624b13b16dc3583"));
}
```

... which would be quite fantastic.